src/include/platform/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -699,6 +699,12 @@ void GenericConfigurationManagerImpl<ConfigClass>::LogDeviceConfig()
         ChipLogProgress(DeviceLayer, "  Product Id: %u (0x%X)", productId, productId);
     }
 
+    {
+        char productName[ConfigurationManager::kMaxProductNameLength + 1];
+        err = deviceInstanceInfoProvider->GetProductName(productName, sizeof(productName));
+        ChipLogProgress(DeviceLayer, "  Product Name: %s", (err == CHIP_NO_ERROR) ? productName : "(not set)");
```

Comment:
Having an error here does not necessarily mean "not set".
Please log an error message containing the error text unless there is a curated list of errors that are expected if missing (e.g. KEY_NOT_FOUND or similar).

If the error would be for example `INSUFFICIENT_MEMORY`, saying 'not set' is incorrect.

Reply:
@andy31415 Can you help to review this? Thanks.